test/benchmark_http_client_test.cc

```diff
@@ -41,41 +41,70 @@ class BenchmarkClientHttpTest : public Test {
 
     auto& tracer = static_cast<Envoy::Tracing::MockHttpTracer&>(*http_tracer_);
     EXPECT_CALL(tracer, startSpan_(_, _, _, _))
-        .WillRepeatedly(
-            Invoke([&](const Envoy::Tracing::Config& config, const Envoy::Http::HeaderMap&,
-                       const Envoy::StreamInfo::StreamInfo&,
-                       const Envoy::Tracing::Decision) -> Envoy::Tracing::Span* {
-              EXPECT_EQ(Envoy::Tracing::OperationName::Egress, config.operationName());
-              auto* span = new NiceMock<Envoy::Tracing::MockSpan>();
-              return span;
-            }));
-    request_generator_ = []() {
+        .WillRepeatedly([&](const Envoy::Tracing::Config& config, const Envoy::Http::HeaderMap&,
+                            const Envoy::StreamInfo::StreamInfo&,
+                            const Envoy::Tracing::Decision) -> Envoy::Tracing::Span* {
+          EXPECT_EQ(Envoy::Tracing::OperationName::Egress, config.operationName());
+          auto* span = new NiceMock<Envoy::Tracing::MockSpan>();
+          return span;
+        });
+  }
+  //Default function for request generator when the content doesn't matter.
+  RequestGenerator getDefaultRequestGenerator() const {
+    RequestGenerator request_generator = []() {
       auto header = std::make_shared<Envoy::Http::TestRequestHeaderMapImpl>(
           std::initializer_list<std::pair<std::string, std::string>>(
               {{":scheme", "http"}, {":method", "GET"}, {":path", "/"}, {":host", "localhost"}}));
       return std::make_unique<RequestImpl>(header);
     };
+    return request_generator;
   }
-
-  void testBasicFunctionality(const uint64_t max_pending, const uint64_t connection_limit,
-                              const uint64_t amount_of_request) {
+  /// Helper function to get properties in a map that should be verified during the test
+  absl::flat_hash_map<std::string, std::string>
+  getTestRecordedProperties(const Envoy::Http::RequestHeaderMap& header) {
+    absl::flat_hash_map<std::string, std::string> properties_map;
+    properties_map["uri"] = std::string(header.getPathValue());
+    return properties_map;
+  }
+  /// Used to set up benchmarkclient. Especially from within the testBenchmarkClientFunctionality.
```

Comment:
nit: one line comments should only start with // not ///.

Reply:
done